### PR TITLE
engine(dm): add config/binlog/binlog-schema command

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #5504 

### What is changed and how it works?
- add config command
- add binlog command
- add binlog-schema command

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by GMHDBJD** [comment]
waiting for #5778

**Review by lance6716 — COMMENTED** [review]
will review test soon

File: engine/executor/dm/unitholder_test.go [hunk 1]
```diff
@@ -124,6 +125,47 @@ func TestUnitHolder(t *testing.T) {
...
+
+	// wrong type
+	msg, err := unitHolder.Binlog(context.Background(), &dmpkg.BinlogTaskRequest{})
+	require.Error(t, err)
+	require.Equal(t, "", msg)
+	// no binlog error
+	unitHolder.unit = syncer.NewSyncer(&config.SubTaskConfig{Flavor: mysql.MySQLFlavor}, nil, nil)
+	unitHolder.runCtx = context.Background()
+	msg, err = unitHolder.Binlog(context.Background(), &dmpkg.BinlogTaskRequest{})
+	require.Error(t, err)
```
Comment (lance6716) [inline engine/executor/dm/unitholder_test.go:140]: reviewer is not easy to understand which error it is. Maybe check error string

File: engine/executor/dm/unitholder_test.go [hunk 2]
```diff
@@ -124,6 +125,47 @@ func TestUnitHolder(t *testing.T) {
...
+
+	// wrong type
+	msg, err := unitHolder.BinlogSchema(context.Background(), &dmpkg.BinlogSchemaTaskRequest{})
+	require.Error(t, err)
+	require.Equal(t, "", msg)
+	// wrong stage
+	unitHolder.unit = syncer.NewSyncer(&config.SubTaskConfig{Flavor: mysql.MySQLFlavor}, nil, nil)
+	unitHolder.runCtx = context.Background()
+	msg, err = unitHolder.BinlogSchema(context.Background(), &dmpkg.BinlogSchemaTaskRequest{})
+	require.Error(t, err)
```
Comment (lance6716) [inline engine/executor/dm/unitholder_test.go:160]: and here

File: engine/jobmaster/dm/config/config_test.go [hunk 3]
```diff
@@ -76,6 +76,14 @@ func TestTaskCfg(t *testing.T) {
 	require.NoError(t, jobCfg.DecodeFile(jobTemplatePath))
 
 	taskCfgs := jobCfg.ToTaskCfgs()
+
+	taskCfgList := make([]*TaskCfg, 0, len(taskCfgs))
+	for _, taskCfg := range taskCfgs {
+		taskCfgList = append(taskCfgList, taskCfg)
+	}
+	jobCfg2 := FromTaskCfgs(taskCfgList)
+	taskCfgs = jobCfg2.ToTaskCfgs()
```
Comment (lance6716) [inline engine/jobmaster/dm/config/config_test.go:85]: what's the purpose of these lines?

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: engine/jobmaster/dm/config/config_test.go (hunk 3 above)
Comment (GMHDBJD) [reply engine/jobmaster/dm/config/config_test.go:85]: make sure there is no change after convert jobCfg to taskCfg and convert it back.

File: engine/executor/dm/unitholder_test.go [hunk 4]
```diff
@@ -124,6 +125,47 @@ func TestUnitHolder(t *testing.T) {
...
+
+	// wrong type
+	msg, err := unitHolder.BinlogSchema(context.Background(), &dmpkg.BinlogSchemaTaskRequest{})
+	require.Error(t, err)
+	require.Equal(t, "", msg)
+	// wrong stage
+	unitHolder.unit = syncer.NewSyncer(&config.SubTaskConfig{Flavor: mysql.MySQLFlavor}, nil, nil)
+	unitHolder.runCtx = context.Background()
+	msg, err = unitHolder.BinlogSchema(context.Background(), &dmpkg.BinlogSchemaTaskRequest{})
+	require.EqualError(t, err, fmt.Sprintf("current stage is %d but not paused, invalid", metadata.StageRunning))
```
Comment (lance6716) [inline engine/executor/dm/unitholder_test.go:160]: maybe we can add String for the stage later.

And message can be "invalid request, current stage is %d, expected stage is paused"

File: engine/jobmaster/dm/config/config_test.go (hunk 3 above)
Comment (lance6716) [reply engine/jobmaster/dm/config/config_test.go:85]: Do you forget to check jobCfg2 equals jobCfg?

**Review by lance6716 — APPROVED** [review]

**Comment by GMHDBJD** [comment]
> Do you forget to check jobCfg2 equals jobCfg?

**Comment by GMHDBJD** [comment]
cc @amyangfei

**Pull request merged**